opt_einsum/path_random.py

```diff
@@ -372,3 +372,6 @@ def random_greedy(inputs, output, idx_dict, memory_limit=None, **optimizer_kwarg
     """
     optimizer = RandomGreedy(**optimizer_kwargs)
     return optimizer(inputs, output, idx_dict, memory_limit)
+
+
+paths.add_path_fn('random-greedy', random_greedy)
```

Comment:
I do like the registration, but the potential circular imports worry me a bit. What do you think of imports the path finders into `paths.py` and register them there?

Reply:
So the registration was my way of getting round the circular import problem since ``path_random`` depends on ``paths`` quite a bit. However switching to ``from . import paths`` in this file might have sorted that already. Alternatively, could perform the registration in the top ``__init__``?

Reply:
The `from . imports paths` did prevent the circular depends in this particular case, but is quite vulnerable to problems in the future. Perhaps we could split off registration into another file or as you said, add it to the init.